vavr/src/main/java/io/vavr/collection/AbstractMultimap.java

```diff
@@ -254,39 +254,63 @@ public M filter(Predicate<? super Tuple2<K, V>> predicate) {
     }
 
     @Override
+    public M reject(Predicate<? super Tuple2<K, V>> predicate) {
+        Objects.requireNonNull(predicate, "predicate is null");
+        return filter(predicate.negate());
+    }
+
+    @Override
     public M filter(BiPredicate<? super K, ? super V> predicate) {
         Objects.requireNonNull(predicate, "predicate is null");
         return filter(t -> predicate.test(t._1, t._2));
     }
 
     @Override
+    public M reject(BiPredicate<? super K, ? super V> predicate) {
+        Objects.requireNonNull(predicate, "predicate is null");
+        return reject(t -> predicate.test(t._1, t._2));
+    }
+
+    @Override
     public M filterKeys(Predicate<? super K> predicate) {
         Objects.requireNonNull(predicate, "predicate is null");
         return filter(t -> predicate.test(t._1));
     }
 
     @Override
+    public M rejectKeys(Predicate<? super K> predicate) {
+        Objects.requireNonNull(predicate, "predicate is null");
+        return reject(t -> predicate.test(t._1));
+    }
+
+    @Override
     public M filterValues(Predicate<? super V> predicate) {
         Objects.requireNonNull(predicate, "predicate is null");
         return filter(t -> predicate.test(t._2));
     }
 
     @Override
-    public M removeAll(BiPredicate<? super K, ? super V> predicate) {
+    public M rejectValues(Predicate<? super V> predicate) {
         Objects.requireNonNull(predicate, "predicate is null");
-        return filter(predicate.negate());
+        return reject(t -> predicate.test(t._2));
+    }
+
+    @Override
+    @Deprecated
+    public M removeAll(BiPredicate<? super K, ? super V> predicate) {
+        return reject(predicate);
     }
 
     @Override
+    @Deprecated
     public M removeKeys(Predicate<? super K> predicate) {
-        Objects.requireNonNull(predicate, "predicate is null");
```

Comment:
I believe you should save this checking. This will be more concrete if you will throw an exception in the called method but not in its implementation. This also applies to other deprecated methods. There are many methods in Vavr where this rule is not respected, but I believe that it should